Completed all the desired features.

jorgecuesta opened the pull request:
1. Full validation of config.json
2. Integration of WTS api using graphql generated
3. Added docker and docker-compose
4. Added github ci to generate the images
5. Updated README.md with more details about this

jorgecuesta committed:
Completed all the desired features.

1. Full validation of config.json
2. Integration of WTS api using graphql generated
3. Added docker and docker-compose
4. Added github ci to generate the images
5. Updated README.md with more details about this

jorgecuesta committed:
Decode config.json from base64 in CI workflow

Previously, the config.json was directly using the secrets.CONFIG_FILE value. This change decodes the base64-encoded CONFIG_FILE before creating config.json, ensuring proper file format and content integrity.

jorgecuesta committed:
Minor enhancements.

* Improved logging
* Added CONFIG_FILE env to override default path
* Fixed retry logic

jorgecuesta committed:
Update README with Docker image build info

Added information about an existing Docker image build available on Docker Hub. This provides clarity on how users can access pre-built images from our CI, facilitating easier deployment.

Alann27 commented on config.go:
```diff
@@ -2,127 +2,372 @@ package wtsc
+	Logger.Info().Str("path", configPath).Msg("reading config file")
 	var jsonFile *os.File
 	var bz []byte
 	if _, err := os.Stat(configPath); err != nil && os.IsNotExist(err) {
```
Shouldn't it log a Fatal If the error is not a `os.IsNotExist` too?

jorgecuesta replied:
Logger.Fatal is there.

Alann27 commented on config.go:
```diff
@@ -2,127 +2,372 @@ package wtsc
-		log.Fatal().Str("configPath", configPath).Err(err).Msg("failed to unmarshal config file")
-	}
-
-	err = ValidateConfig(&cfg)
```
Why did you remove the ValidateConfig?

jorgecuesta replied:
was moved out on every place it is called, but now I move it at the very end of LoadConfig, previously was not trigger Fatal() on hot reload of config file to prevent process stop due to a bad change.

Alann27 commented on README.md:
```diff
@@ -1,19 +1,71 @@
+  "tx_memo": "wtsc", // Transaction memo
+  "tx_fee": 10000, // Transaction fee
+  "domain": "example.com", // Your nodes domain (e.g., "poktscan.cloud" or "c0d3r.org")
+  "chain_pool": ["0021", "0003"], // Chain IDs available in your fleet
+  "servicer_keys": ["PRIVATE_KEY"], // List of private keys for signing stake transactions
+  "stake_weight": 4, // Used by the "What to Stake" service to estimate potential rewards (1-4)
+  "min_increase_percent": 5, // Minimum percentage increase expected to process stakes
+  "min_service_stake": [ // Minimum number of nodes for specific services
+    {
+      "service": "0021",
+      "min_node": 1
+    }
+  ],
```
`chain_pool` should not be named `service_pool`? Because in the code, chain is always named service and I think is better to just have one name for the same thing.

jorgecuesta committed:
Refactor config validation and update terminology

Removed redundant config validation calls and renamed 'chain_pool' to 'service_pool' for clarity. Updated documentation and config samples to reflect these changes. Ensured consistency across codebase and README.md for better understanding and maintenance.

jorgecuesta committed:
Remove pocket-core module references of .idea folder.

Alann27 commented on README.md:
```diff
@@ -27,42 +27,39 @@ Before using the provided recommendations, it's important to understand several
+| tx_memo               | string              | Transaction memo                                                                                                 |
+| tx_fee                | integer             | Transaction fee                                                                                                  |
+| domain                | string              | Your node's domain (e.g., "poktscan.cloud" or "c0d3r.org")                                                       |
+| chain_pool            | array of strings    | Chain IDs available in your fleet                                                                                |
```
chain_pool still here, replace it with service_pool

jorgecuesta committed:
Refactor validation and enhance documentation

Updated IsValidServicerList to support dry mode and revised associated validation code. Enhanced README with a table of contents and detailed usage instructions, including dry mode recommendations. Renamed output files for clarity, and added issue and pull request templates to streamline contributions.
